test/callback_api.js

```diff
@@ -210,6 +214,34 @@ suite('sending messages', function() {
     });
   });
 
+  channel_test('saturate buffer', function(ch, done) {
+    var msg = randomString();
+    ch.assertQueue('', {exclusive: true}, function(e, q) {
+      if (e !== null) return done(e);
+      let ok;
+      for (let i = 0; i < 2047; i++) {
```

Comment:
This might need to be higher on different spec machines. If it becomes flakey I'll remove the test, but at least it demonstrated the feature is working locally.